src/hidra/receiver/asapo_transfer.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+import random
+import signal
+import socket
+from threading import Event
+from time import sleep
+import yaml
+import zmq
+
+from hidra import Transfer, __version__, generate_filepath, _constants
+from plugins.asapo_producer import AsapoWorker
+
+logger = logging.getLogger(__name__)
+
+
+class Stopped(Exception):
+    """Raised when a run is stopped."""
+    pass
+
+
+class TransferConfig:
+    def __init__(self, signal_host, target_host, detector_id,
+                 endpoint, beamline, default_data_source,
+                 token_path='/gpfs/asapo/shared/beamline_tokens',
+                 target_dir='', n_threads=1, start_file_idx=1, beamtime='auto',
+                 file_regex="current/raw/(?P<scan_id>.*)_(?P<file_idx_in_scan>.*).h5",
+                 timeout=30, reconnect_timeout=3, log_level="INFO"):
+        
+        self.detector_id = detector_id
+        self.log_level = log_level
+        
+        self.endpoint = endpoint
+        self.beamtime = beamtime
+        self.beamline = beamline
+        self.token = read_token(token_path, beamline)
```

Comment:
I would avoid large side effects like reading a file in the constructor. Do this in `construct_config` and just take the token as an argument here.

Reply:
Done.